rep-2009.rst

```diff
@@ -0,0 +1,468 @@
+REP: 2009
+Title: Type Negotiation Feature
+Author: Audrow Nash <audrow@openrobotics.org>, Chris Lalancette <clalancette@openrobotics.org>, Gonzalo de Pedro <gonzalodepedro@ekumenlabs.com>, William Woodall <william@openrobotics.org>
+Status: Accepted
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 04-Nov-2021
+Post-History:
+
+Abstract
+========
+
+This REP proposes an extension to ROS 2 Lifecycle Nodes that will make it possible to negotiate the message types used by publishers and subscribers.
+
+
+Motivation
+==========
+
+The primary reason for this change is to allow nodes to publish different types of messages that better allow the system to optimize its behavior.
```

Comment:
The big change here is to allow multiple message types per topic, correct?

If so: this sentence should clearly state that. Something like:
```suggestion
The primary reason for this change is to allow nodes to publish different types of messages on a single topic that better allow the system to optimize its behavior.
```